app/models/sample.rb

```diff
@@ -112,6 +111,18 @@ def input_files_checks
     end
   end
 
+  def required_metadata_fields
+    host_genome.metadata_fields.where(is_required: 1).pluck(:name)
+  end
+
+  def missing_required_metadata_fields
+    required_metadata_fields - metadata.pluck(:key)
```

Comment:
since we want to remove the :key on Metadatum later could you do `required_metadata_fields - metadata.map(&:metadata_field).pluck(:key)`